addons/mail/static/src/new/thread/message.js

```diff
@@ -80,6 +80,22 @@ export class Message extends Component {
         return this.message.resId === (thread.resId || this.props.threadId);
     }
 
+    get isSubjectSimilarToOriginThreadName() {
+        if (!this.message.subject || !this.originThread || !this.originThread.name) {
+            return false;
+        }
+        const prefixList = ["re", "fw", "fwd"];
+        // match any prefix as many times as possible
+        const prefixesRegexp = new RegExp(`^((${prefixList.join("|")}):\\s*)+`);
+        const cleanedSubject = this.message.subject.toLowerCase().replace(prefixesRegexp, "");
+        const cleanedOriginThreadName = this.originThread.name.toLowerCase().trim();
+        return cleanedSubject === cleanedOriginThreadName;
```

Comment:
If you think the code is better this way it would be nice to make this change in master so that diff in this PR is reduced to seen the code moved.

+ adding tests to cover edge cases to make sure the change is keeping the behavior